File: ControlAssuranceAPI/Models/IAPAction.cs
using System;
using System.Collections.Generic;

namespace CAT.Models;

public partial class IAPAction
{
    public int ID { get; set; }

    public string? Title { get; set; }

    public string? Details { get; set; }

    public int? IAPPriorityId { get; set; }

    public int? IAPStatusTypeId { get; set; }

    public int? IAPTypeId { get; set; }

    public int? CreatedById { get; set; }

    public DateTime? CreatedOn { get; set; }

    public string? Attachment { get; set; }

    public bool? IsLink { get; set; }

    public DateTime? OriginalCompletionDate { get; set; }

    public DateTime? CompletionDate { get; set; }

    public bool? MonthlyUpdateRequired { get; set; }

    public bool? MonthlyUpdateRequiredIfNotCompleted { get; set; }

    public bool? IsArchive { get; set; }

    public int? ParentId { get; set; }

    public int? GroupNum { get; set; }

    public string? ActionLinks { get; set; }

    public virtual User? CreatedBy { get; set; }

    public virtual ICollection<IAPActionDirectorate> IAPActionDirectorates { get; } = new List<IAPActionDirectorate>();

    public virtual ICollection<IAPActionUpdate> IAPActionUpdates { get; } = new List<IAPActionUpdate>();

    public virtual ICollection<IAPAssignment> IAPAssignments { get; } = new List<IAPAssignment>();

    public virtual IAPPriority? IAPPriority { get; set; }

    public virtual IAPStatusType? IAPStatusType { get; set; }

    public virtual IAPType? IAPType { get; set; }
}

public class IAPActionView_Result
{
    public string? ID { get; set; }
    public string? Title { get; set; }
    public string? Priority { get; set; }
    public string? Status { get; set; }
    public string? CreatedBy { get; set; }
    public int CreatedById { get; set; }
    public string? CreatedOn { get; set; }
    public string? ActionOwners { get; set; }
    public string? OwnerIds { get; set; }
    public string? Update { get; set; }
    public string? Directorates { get; set; }
    public int IAPTypeId { get; set; }
    public string? Type { get; set; }
    public int CurrentPeriodId { get; set; }
    public string? CompletionDate { get; set; }

}

File: control-assurance-sp/src/types/IAPAction.ts
import { IIAPActionDirectorate } from ".";
import { IEntity } from "./Entity";

export interface IIAPAction  extends IEntity {

    Details?: string;
    IAPPriorityId?:number;
    IAPStatusTypeId?:number;
    IAPTypeId?:number;
    CreatedById?:number;
    CreatedOn?:Date; 
    Attachment?:string;
    IsLink?:boolean;
    CompletionDate?:Date;
    OriginalCompletionDate?:Date;
    MonthlyUpdateRequired?:boolean;
    MonthlyUpdateRequiredIfNotCompleted?:boolean;
    ActionLinks?: string;
    IsArchive?:boolean;
    IAPActionDirectorates?: IIAPActionDirectorate[];
}

export class IAPAction implements IIAPAction{ 
    public ID: number = 0;
    public Title: string = null;
    public Details?: string = null;
    public IAPPriorityId?:number = null;
    public IAPStatusTypeId?:number = null;
    public IAPTypeId?:number = null;
    public CreatedById?:number = null;
    public CreatedOn?:Date = null;
    public Attachment?:string = null;
    public IsLink:boolean = null;
    public CompletionDate = null;
    public OriginalCompletionDate = null;
    public MonthlyUpdateRequired = false;
    public MonthlyUpdateRequiredIfNotCompleted = false;
    public ActionLinks = null;
    public IsArchive?:boolean = false;
    public IAPActionDirectorates = [];

    constructor(IAPPriorityId: number, IAPStatusTypeId:number, IAPTypeId: number) {
        this.IAPPriorityId = IAPPriorityId;
        this.IAPStatusTypeId = IAPStatusTypeId;
        this.IAPTypeId = IAPTypeId;
    }
}